Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,385 @@
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPRuntimeException;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\PhpAmqpLibQueueManager;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueManager;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+use Smartbox\Integration\FrameworkBundle\Service;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+/**
+ * Class PhpAmqpLibDriver
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers
+ */
+class PhpAmqpLibDriver extends Service implements PurgeableQueueDriverInterface
+{
+    use UsesSerializer;
+    use UsesExceptionHandlerTrait;
+    use ContainerAwareTrait;
+    
+    /**
+     * Maximum amount of time (in seconds) to wait for a message.
+     */
+    const WAIT_TIMEOUT = 10;
+
+    /**
+     * Maximum amount of time (in seconds) to set as an interval while consuming messages
+     */
+    const WAIT_PERIOD = 0.2;
+
+    /**
+     * @var \AMQPQueue
+     */
+    private $queue;
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueingTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var PhpAmqpLibQueueManager
+     */
+    private $manager;
+
+    /**
+     * @var string Host form Message Broker
+     */
+    private $host;
+
+    /**
+     * @var string Username to connect to Message broker
+     */
+    private $username;
+
+    /**
+     * @var string $password
+     */
+    private $password;
+
+    /**
+     * @var string $port
+     */
+    private $port;
+
+    /**
+     * @var string $vhost
+     */
+    private $vhost;
+
+    /**
+     * @var array
+     */
+    private $connectionId = [];
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * PhpAmqpLibDriver constructor.
+     * @param QueueManager $manager
+     */
+    public function __construct(PhpAmqpLibQueueManager $manager)
+    {
+        parent::__construct();
+        $this->manager = $manager;
+        $this->format = self::FORMAT_JSON;
+    }
+
+    /**
+     * PhpAmqpLibDriver destruct
+     */
+    public function __destruct()
+    {
+        $this->manager->disconnect();
+    }
+
+    /**
+     * @param $host
+     * @param string $username
+     * @param string $password
+     * @param string $format
+     * @param null $port
+     * @param null $vhost
+     */
+    public function configure($host, $username, $password, $format = self::FORMAT_JSON, $port = null, $vhost = null)
```

Comment:
In the interface that defines this method ```Components/Queues/Drivers/QueueDriverInterface.php``` there is no the arguments ```port``` and ```vhost```

Reply:
Ok, I removed the parameters once tested and verified that there is no impact int the functionality.